libs/hdf-converters/src/veracode-mapper.ts

```diff
@@ -0,0 +1,576 @@
+    _.set(parsedXML, `${SCA_CVES}${m}].impact`, impact);
+    for (
+      let l = 0;
+      l <
+      Number(
+        _.get(
+          parsedXML,
+          'detailedreport.software_composition_analysis.vulnerabilities.length'
+        )
+      );
+      l++
+    ) {
+      let hascve = false;
+      for (
+        let n = 0;
+        n < Number(_.get(parsedXML, `${SCA_VULNERABILITIES}${l}].cves.length`));
+        n++
+      ) {
+        const cve = _.get(parsedXML, `${SCA_VULNERABILITIES}${l}].cves[${n}]`);
+        if (cve === currcve) {
+          hascve = true;
+          location +=
+            ' ' +
+            _.get(
+              parsedXML,
+              `${SCA_VULNERABILITIES}${l}].file_paths.file_path`
+            );
+          _.set(parsedXML, `${SCA_CVES}${m}].paths`, location);
+        }
+      }
+      if (hascve === true) {
+        const proxy = _.cloneDeep(parsedXML);
+        let omitted = _.omit(
+          proxy,
+          `${SCA_VULNERABILITIES}${l}].vulnerabilities`
+        );
+        omitted = _.omit(omitted, `${SCA_VULNERABILITIES}${l}].cves`);
+        components.push(_.get(omitted, `${SCA_VULNERABILITIES}${l}]`));
+      }
+    }
+    _.set(parsedXML, `${SCA_CVES}${m}].components`, components);
+  }
+  return parsedXML;
+}
+
+function cvesformatter(parsedXML: any) {
+  const vulnarr = [];
+  for (
+    let k = 0;
+    k <
+    Number(
+      _.get(
+        parsedXML,
+        'detailedreport.software_composition_analysis.vulnerabilities.length'
+      )
+    );
+    k++
+  ) {
+    const cves = [];
+    for (
+      let l = 0;
+      l <
+      Number(
+        _.get(
+          parsedXML,
+          `${SCA_VULNERABILITIES}${k}].vulnerabilities.vulnerability.length`
+        )
+      );
+      l++
+    ) {
+      cves.push(
+        _.get(
+          parsedXML,
+          `${SCA_VULNERABILITIES}${k}].vulnerabilities.vulnerability[${l}].cve_id`
+        )
+      );
+      vulnarr.push(
+        _.get(
+          parsedXML,
+          `${SCA_VULNERABILITIES}${k}].vulnerabilities.vulnerability[${l}]`
+        )
+      );
+    }
+    const cvei = _.get(
+      parsedXML,
+      `${SCA_VULNERABILITIES}${k}].vulnerabilities.vulnerability.cve_id`
+    );
+    if (
+      _.has(
+        parsedXML,
+        `${SCA_VULNERABILITIES}${k}].vulnerabilities.vulnerability.cve_id`
+      )
+    ) {
+      cves.push(cvei);
+      vulnarr.push(
+        _.get(
+          parsedXML,
+          `${SCA_VULNERABILITIES}${k}].vulnerabilities.vulnerability`
+        )
+      );
+    }
+    _.set(parsedXML, `${SCA_VULNERABILITIES}${k}].cves`, cves);
+  }
+  _.set(
+    parsedXML,
+    'detailedreport.software_composition_analysis.cves',
+    vulnarr
+  );
+  return parsedXML;
+}
+function parseXml(xml: string) {
+  const options = {
+    attributeNamePrefix: '',
+    textNodeName: 'text',
+    ignoreAttributes: false
+  };
+
+  const parsedXML = parser.parse(xml, options);
```

Comment:
ok on further review, i think doing any of this pre-processing like was done here is more or less wrong / lends itself to bad practices and unreadable code.

the functional issue is that some attributes are a bit harder to get to like staticflaws.flaws and some sections require a flip (components -> vulnerabilities/cves => cves -> components).  i've already detailed how to do the flaws one.  the other one is not too bad either because of the `arrayTransform` - see the `consolidate` function in the asffmapper but basically you parse the file as is and generate an array that'll look like in a very abstract sense `[{cve: info for the first cve, componentsAffected: [only the first component]}, ..., {cve: info for the mth cve, componentsAffected: [only the first component]}, ..., {cve: info for the first cve, componentsAffected: [only the nth component]}, ..., {cve: info for the mth cve, componentsAffected: [only the nth component]}]` and then you reduce/consolidate this array based off of the cve's id to create an array that looks like `[{cve: info for the first cve, componentsAffected: [first component, ..., nth component]}, ..., {cve: mth cve, componentsAffected: [first component, ..., nth component]}]`.  it sounds more complicated than it actually is.